src/main/scala/rise/Cuda/TypedDSL.scala

```diff
@@ -0,0 +1,51 @@
+package rise.Cuda
+
+import rise.core.DSL._
+import rise.core.semantics.HalfData
+import rise.core.{DepApp, Expr, Literal, Primitive}
+import rise.core.types.Nat
+
+object TypedDSL {
+  object mapBlock {
+    def apply(): ToBeTyped[Primitive] = primitives.mapBlock('x')
+    def apply[T <: Expr](e: ToBeTyped[T]): ToBeTyped[rise.core.App] = primitives.mapBlock('x')(e)
+    def apply(dim: Char): ToBeTyped[Primitive] = primitives.mapBlock(dim)
+  }
+
+  object mapGlobal {
+    def apply(): ToBeTyped[Primitive] = primitives.mapGlobal('x')
+    def apply[T <: Expr](e: ToBeTyped[T]): ToBeTyped[rise.core.App] = primitives.mapGlobal('x')(e)
+    def apply(dim: Char): ToBeTyped[Primitive] = primitives.mapGlobal(dim)
+  }
+
+  object mapThreads {
+    def apply(): ToBeTyped[Primitive] = primitives.mapThreads('x')
+    def apply[T <: Expr](e: ToBeTyped[T]): ToBeTyped[rise.core.App] = primitives.mapThreads('x')(e)
+    def apply(dim: Char): ToBeTyped[Primitive] = primitives.mapThreads(dim)
+  }
```

Comment:
Are `mapBlock` and `mapThreads` doing something different than `mapGlobal`/`mapLocal`/`mapWorkGroup`? It would be good to merge these primitives with the OpenCL ones in a common abstraction.

Reply:
mapBlock is like mapWorkgroup and mapThreads like mapLocal. mapGlobal exists in both. Additionally are mapWarp and mapLane in CUDA available.
Common abstractions make sense

Reply:
mentioned in #106